apps/ledger-live-desktop/tests/specs/speculos/send.tx.spec.ts

```diff
@@ -1,48 +1,265 @@
+      await app.send.fillRecipient(tokenTransactionValid.accountToCredit.address);
+      await app.send.checkContinueButtonEnable();
+      await app.layout.checkInputErrorNotVisible();
+    },
+  );
+});
+
+test.describe("Check invalid address input error", () => {
+  test.use({
+    userdata: "speculos-tests-app",
+    testName: `Invalid_address_input_error_${transactionInvalidAddress.accountToDebit.currency.uiName}`,
+    speculosCurrency:
+      specs[transactionInvalidAddress.accountToDebit.currency.deviceLabel.replace(/ /g, "_")],
+    speculosOffset: 0,
+  });
+
+  test(
+    `Check [${transactionInvalidAddress.accountToDebit.currency}] invalid address input error`,
+    {
+      annotation: {
+        type: "TMS",
+        description: "B2CQA-1904",
+      },
+    },
+    async ({ app }) => {
+      await app.layout.goToAccounts();
+      await app.accounts.navigateToAccountByName(
+        transactionInvalidAddress.accountToDebit.accountName,
+      );
+
+      await app.account.clickSend();
+      await app.send.fillRecipient(transactionInvalidAddress.accountToCredit.address);
+      await app.send.checkInvalidAddressError(transactionInvalidAddress);
+    },
+  );
+});
+
+for (const [i, transaction] of transactionsInputsInvalid.entries()) {
+  test.describe("Check invalid amount input error", () => {
     test.use({
       userdata: "speculos-tests-app",
-      testName: `sendApprove_${transaction.accountToDebit.currency.uiName}`,
-      speculosCurrency: specs[transaction.accountToDebit.currency.deviceLabel.replace(/ /g, "_")],
+      testName: `Invalid_amount_input_error_${transaction.transaction.accountToDebit.currency.uiName}`,
+      speculosCurrency:
+        specs[transaction.transaction.accountToDebit.currency.deviceLabel.replace(/ /g, "_")],
       speculosOffset: i,
     });
 
     test(
-      `[${transaction.accountToDebit.accountName}] send Approve`,
+      `Check [${transaction.expectedErrorMessage}] for [${transaction.transaction.accountToDebit.currency.uiName}] - invalid amount input error - ${i} `,
       {
         annotation: {
           type: "TMS",
           description: "B2CQA-473",
         },
       },
       async ({ app }) => {
-        await addTmsLink(getDescription(test.info().annotations).split(", "));
+        await app.layout.goToAccounts();
+        await app.accounts.navigateToAccountByName(
+          transaction.transaction.accountToDebit.accountName,
+        );
+
+        await app.account.clickSend();
+        await app.send.fillRecipient(transaction.transaction.accountToCredit.address);
+        await app.send.clickContinue();
+        await app.send.fillAmount(transaction.transaction.amount);
+        await app.send.checkContinueButtonDisabled();
+        await app.layout.checkErrorMessage(transaction.expectedErrorMessage);
+      },
+    );
+  });
+}
+
+for (const [i, transaction] of transactionInputValid.entries()) {
+  test.describe("Check valid amount input", () => {
```

Comment:
why do we need this separate test if we already have sunny send test case (first one in the file)? can't we simply add those two last verifications to already existing test?

Reply:
The first is a total e2e test and the broadcast is done at the end (using Sepolia). This test checks that the Send Max button and the Continue button are working, and that we can enter an amount less than the balance, which is already tested in the first test (I can delete this part if you want but we will have to update the Jira ticket). If I add the possibility to do a send max to the first one, we will use all the available balance, which is why they are separated.